lib/datadog/statsd.rb

```diff
@@ -213,6 +213,26 @@ def distribution(stat, value, opts = EMPTY_OPTIONS)
       send_stats(stat, value, DISTRIBUTION_TYPE, opts)
     end
 
+    # Reports execution time of the provided block as a distribution.
+    #
+    # If the block fails, the stat is still reported, then the error
+    # is reraised
+    #
+    # @param [String] stat stat name.
+    # @param [Numeric] value distribution value.
+    # @param [Hash] opts the options to create the metric with
+    # @option opts [Numeric] :sample_rate sample rate, 1 for always
+    # @option opts [Array<String>] :tags An array of tags
+    # @example Report the time (in ms) taken to activate an account
+    #   $statsd.distribution_time('account.activate') { @account.activate! }
+    def distribution_time(stat, opts = EMPTY_OPTIONS)
+      opts = { sample_rate: opts } if opts.is_a?(Numeric)
```

Comment:
I added this while adding specs, to mimic what happens in the `timing` method below. Is that what you'd like to do here?

Reply:
I suspect this is for backward-compatibility, so can probably be omitted here.